_posts/2018-03-29-Rust-1.25.md

```diff
@@ -0,0 +1,158 @@
+being a step closer to AVR support.
+
+A new way to write `use` statements has landed: [nested import groups]. If you've
+ever written a set of imports like this:
+
+```rust
+use std::fs::File;
+use std::io::Read;
+use std::path::{Path, PathBuf};
+```
+
+You can now write this:
+
+```rust
+// on one line
+use std::{fs::File, io::Read, path::{Path, PathBuf}};
+
+// with some more breathing room
+use std::{
+    fs::File,
+    io::Read,
+    path::{
+        Path,
+        PathBuf
+    }
+};
+```
+
+This can reduce some repetition, and make things a bit more clear.
+
+There are two big documentation changes in this release: first, [Rust By
+Example is now included on doc.rust-lang.org]! We'll be redirecting the old
+domain there shortly. We hope this will bring more attention to a great
+resource, and you'll get a local copy with your local documentation.
+
+Second, back in Rust 1.23, we talked about the change from Hoedown to
+pulldown-cmark. In Rust 1.25, pulldown-cmark is now the default rendering.
+After years, we have finally removed the last bit of C from rustdoc, and
+properly follow the CommonMark spec.
+
+Finally, in [RFC 1358], `#[repr(align(x))]` was accepted. In Rust
+1.25, [it is now stable]! This attribute lets you set the [alignment]
+of your `struct`s:
+
+```rust
+struct NotAligned(i32);
+
+assert_eq!(std::mem::align_of::<NotAligned>(), 4);
+assert_eq!(std::mem::size_of::<NotAligned>(), 4);
+
+#[repr(align(16))]
+struct Align16(i32);
+
+assert_eq!(std::mem::align_of::<Align16>(), 16);
+assert_eq!(std::mem::size_of::<Align16>(), 16);
+```
+
+If you're working with low-level stuff, control of these kinds of things
+can be very important!
+
+[upgraded to LLVM 6]: https://github.com/rust-lang/rust/pull/47828
+[nested import groups]: https://github.com/rust-lang/rust/pull/47948
+[Rust By Example is now included on doc.rust-lang.org]: https://doc.rust-lang.org/rust-by-example/
+[RFC 1358]: https://github.com/rust-lang/rfcs/blob/master/text/1358-repr-align.md
+[it is now stable]: https://github.com/rust-lang/rust/pull/47006
+[alignment]: https://en.wikipedia.org/wiki/Data_structure_alignment
+
+
+See the [detailed release notes][notes] for more.
+
+### Library stabilizations
+
+The biggest story in libraries this release is [`std::ptr::NonNull<T>`]. This type
+is similar to `*mut T`, but is non-null and covariant. This blog post isn't the right
+place to explain variance, but in a nutshell, `NonNull<T>`, well, guarantees that it
+won't be null, which means that `Option<NonNull<T>>` is the same size as `Option<T>`.
+If you're building a data structure with unsafe code, `NonNull<T>` is often the right
+type for you!
+
+[`std::ptr::NonNull<T>`]: https://doc.rust-lang.org/std/ptr/struct.NonNull.html
+
+`libcore` has [gained a `time` module](https://doc.rust-lang.org/core/time/),
+containing the `Duration` type previously only available in `libstd`.
+
+Additionally, the `new`, `from_secs`, `from_milis`, `from_micros`, and `from_nanos`
```

Comment:
`from_micros` and `from_nanos` are unstable. Does it make sense to mention them here if you can't use them on stable?

Reply:
i didn't realize this; it was in the changelog